api.planx.uk/pay/index.test.ts

```diff
@@ -0,0 +1,171 @@
+import nock from "nock";
+import { queryMock } from "../tests/graphqlQueryMock";
+import supertest from "supertest";
+
+import app from "../server";
+
+jest.mock("@opensystemslab/planx-core", () => {
+  return {
+    CoreDomainClient: jest.fn().mockImplementation(() => ({
+      session: {
+        findDetails: jest.fn().mockImplementation(() => ({ lockedAt: null }))
+      }
+    })) 
+  }
+});
+
+describe("sending a payment to GOV.UK Pay", () => {
+  const govUKResponse = {
+    amount: 12,
+    reference: "a1234",
+    state: {
+      status: "success",
+      finished: true,
+    },
+    payment_id: "a13345",
+    created_date: "2021-04-30T20:26:34.416Z",
+    _links: {
+      next_url: {
+        href: "https://gov.uk/pay/secret_token",
+      },
+    },
+  };
+
+  beforeEach(() => {
+    nock("https://publicapi.payments.service.gov.uk/v1/payments")
+      .post("")
+      .reply(200, govUKResponse);
+
+    queryMock.mockQuery({
+      name: "InsertPaymentStatus",
+      matchOnVariables: false,
+      data: {},
+    });
+  });
+
+  it("proxies request", async () => {
+    await supertest(app)
+      .post("/pay/southwark?flowId=7cd1c4b4-4229-424f-8d04-c9fdc958ef4e&sessionId=f2d8ca1d-a43b-43ec-b3d9-a9fec63ff19c")
+      .send({
+        amount: 100,
+        reference: "12343543",
+        description: "New application",
+        return_url: "https://editor.planx.uk",
+      })
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toEqual(govUKResponse);
+      });
+  });
+
+  it.skip("succeeds with missing query string", async () => {
```

Comment:
I've skipped this test just to flag it in review - it looks very intentional, but a `sessionId` is going to be required for this API call going forward. I think it's likely that all other query params are also required (they're used for the audit log).

Is there any context around this test or why query params should be optional here @builditben  - is it to do with how proxies are nested for ITP and query params not necessarily passed on?

If so, I'm sure I can do a check for `sessionId` in either body or query params but I just wanted to cross check before making assumptions here.

Reply:
Thanks for the sense check that I'm not missing something silly here - I'll take a closer look 🔍